numpy/lib/twodim_base.py

```diff
@@ -172,21 +245,33 @@ def rot90(m, k=1):
     >>> np.rot90(m, 2)
     array([[4, 3],
            [2, 1]])
+    >>> m = np.arange(8).reshape((2,2,2))
+    >>> np.rot90(m, 1, (1,2))
+    array([[[1, 3],
+            [0, 2]],
+
+          [[5, 7],
+           [4, 6]]])
 
     """
     m = asanyarray(m)
+
     if m.ndim < 2:
         raise ValueError("Input must >= 2-d.")
-    k = k % 4
+
+    k %= 4
+    axes_list = arange(0, m.ndim)
+    axes_list[axes[0]], axes_list[axes[1]] = axes_list[axes[1]], axes_list[axes[0]]
+
     if k == 0:
```

Comment:
You might consider rearranging your if statement so that `axes_list` is only computed if it will be needed:

    if k == 2:
        ...
    else:
        axes_list = ...
        if k == 0:
            ...
        else:
            ...